packages/preferences/src/browser/preferences-widget.ts

```diff
@@ -0,0 +1,237 @@
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import {inject, named} from "inversify";
+import {Message} from "@phosphor/messaging";
+import {PreferencesMenuFactory} from "./preference-menu-factory";
+import {PreferencesDecorator} from "./preferences-decorator";
+import {
+    ContextMenuRenderer,
+    ExpandableTreeNode,
+    PreferenceSchemaProvider,
+    SelectableTreeNode,
+    PreferenceProperty,
+    PreferenceScope,
+    PreferenceService,
+    TreeModel,
+    TreeNode,
+    TreeProps,
+    TreeWidget,
+    WidgetManager,
+    PreferenceProvider,
+    Saveable
+} from "@theia/core/lib/browser";
+import {
+    USER_PREFERENCES_TREE_WIDGET_ID,
+    WORKSPACE_PREFERENCES_TREE_WIDGET_ID
+} from "./preference-frontend-contribution";
+import {SplitPanel} from "@phosphor/widgets/lib/splitpanel";
+import {UserPreferenceProvider} from "./user-preference-provider";
+import {WorkspacePreferenceProvider} from "./workspace-preference-provider";
+import {EditorWidget} from "@theia/editor/lib/browser";
+import {Emitter, Event, MaybePromise} from '@theia/core';
+import {Widget} from "@phosphor/widgets";
+
+export abstract class PreferencesContainer extends SplitPanel implements Saveable {
+
+    dirty: boolean;
+    autoSave: "on" | "off";
+    save: () => MaybePromise<void>;
+    onDirtyChanged: Event<void>;
+    private readonly onDirtyChangedEmitter: Emitter<void>;
+
+    @inject(WidgetManager) protected readonly widgetManager: WidgetManager;
+    private readonly _treeWidgetId: string;
+    private _treeWidget: Widget;
+    private _editorWidget: EditorWidget;
+
+    protected constructor(id: string, label: string, treeWidgetId: string) {
+        super();
+
+        this.id = id;
+        this.title.label = label;
+        this._treeWidgetId = treeWidgetId;
+        this.title.closable = true;
+        this.title.iconClass = 'fa fa-sliders';
+
+        this.dirty = false;
+        this.onDirtyChangedEmitter = new Emitter<void>();
+        this.onDirtyChanged = this.onDirtyChangedEmitter.event;
+        this.save = () => this._editorWidget.saveable.save();
+    }
+
+    abstract async getEditor(): Promise<EditorWidget | undefined>;
+
+    onCloseRequest(msg: Message) {
+        this._treeWidget.close();
+        this._editorWidget.close();
+        this.dirty = false;
+        super.onCloseRequest(msg);
+    }
+
```

Comment:
We need to set the focus (`treewidget.activate`) during `onActivateRequest`.
```
    protected async onActivateRequest(msg: Message): Promise<void> {
        this._treeWidget.activate();
        super.onActivateRequest(msg);
    }
```

Reply:
To clarify: Currently the focus is only set on first attach. If you add the method above it works properly on subsequent activation, too.

Reply:
done